File: 0xxx/01xx/012x/0121.js
/**
 * @param {number[]} prices
 * @return {number}
 */
const maxProfit = function (prices) {
  if (prices.length === 1) return 0
  let buyDay = 0
  let sellDay = 1
  let profit = 0

  while (sellDay < prices.length) {
    if (prices[sellDay] - prices[buyDay] > profit) {
      profit = prices[sellDay] - prices[buyDay]
    } else if (prices[sellDay] < prices[buyDay]) {
      buyDay = sellDay
    }
    sellDay++
  }

  return profit
}

File: 0xxx/01xx/012x/0121.py
class Solution:
  def maxProfit(self, prices: List[int]) -> int:
    buyDay = 0
    sellDay = 1
    profit = 0

    while sellDay < len(prices):
      if prices[sellDay] - prices[buyDay] > profit:
        profit = prices[sellDay] - prices[buyDay]
      elif prices[sellDay] < prices[buyDay]:
        buyDay = sellDay
      sellDay += 1

    return profit
    